### PR TITLE
feat: Introduce OccEndpointService to manage logic for generating endpoint urls, leveraging endpoint configuration

### PR DESCRIPTION
**Pull request by dunqan**
This is a follow-up and more general approach to cofigurability mechanism started in GH-1593.

It is also required for GH-1344

Closes GH-1750

### CODE REVIEW

File: projects/core/src/auth/http-interceptors/client-token.interceptor.ts [hunk 1]
```diff
@@ -33,7 +31,10 @@ export class ClientTokenInterceptor implements HttpInterceptor {
     return this.getClientToken(request).pipe(
       take(1),
       switchMap((token: ClientToken) => {
-        if (token && request.url.indexOf(this.baseReqString) > -1) {
+        if (
+          token &&
```
Comment (tobi-or-not-tobi) [inline projects/core/src/auth/http-interceptors/client-token.interceptor.ts]: What about change `take(1)` into `first(Boolean)` and leave out the first condition?

File: projects/core/src/auth/http-interceptors/user-token.interceptor.ts [hunk 2]
```diff
@@ -32,7 +31,7 @@ export class UserTokenInterceptor implements HttpInterceptor {
   ): Observable<HttpEvent<any>> {
     if (
       this.userToken &&
```
Comment (tobi-or-not-tobi) [inline projects/core/src/auth/http-interceptors/user-token.interceptor.ts]: hard to read (but not your concern ;) ), but shall we move it to a private method that tells us from signature what this is about?

File: projects/core/src/cart/occ/cart.service.ts [hunk 3]
```diff
@@ -29,17 +29,14 @@ const DETAILS_PARAMS =
 
 @Injectable()
 export class OccCartService {
-  constructor(protected http: HttpClient, protected config: OccConfig) {}
+  constructor(
+    protected http: HttpClient,
+    private occEndpoints: OccEndpointsService
+  ) {}
 
   protected getCartEndpoint(userId: string) {
```
Comment (tobi-or-not-tobi) [inline projects/core/src/cart/occ/cart.service.ts]: can we add `:void`

File: projects/core/src/auth/http-interceptors/client-token.interceptor.ts (hunk 1 above)
Comment (dunqan) [reply projects/core/src/auth/http-interceptors/client-token.interceptor.ts]: `first(Boolean)` will make it wait for the token if not present, which we probably don't want here

File: projects/core/src/cart/occ/cart.service.ts (hunk 3 above)
Comment (dunqan) [reply projects/core/src/cart/occ/cart.service.ts]: It actually should be a `string`, thanks, fixed.

File: projects/core/src/auth/http-interceptors/user-token.interceptor.ts (hunk 2 above)
Comment (dunqan) [reply projects/core/src/auth/http-interceptors/user-token.interceptor.ts]: Ok

**Pull request merged**